src/Hedgehog/Extras/Stock/IO/Network/Port.hs

```diff
@@ -0,0 +1,17 @@
+module Hedgehog.Extras.Stock.IO.Network.Port
+  ( randomPort
+  ) where
+
+import           Control.Monad (Monad (..))
+import           Data.Function
+import           Network.Socket
+import           System.IO (IO)
+
+-- | Return a random available port on a specified host address
+randomPort :: HostAddress -> IO PortNumber
+randomPort hostAddress = do
+  sock <- socket AF_INET Stream defaultProtocol
+  bind sock $ SockAddrInet 0 hostAddress
```

Comment:
This works because `0` is akin to https://hackage.haskell.org/package/network-3.2.0.0/docs/Network-Socket.html#v:defaultPort right?

I think this ought to be explained with a comment. Otherwise this code doesn't look like it's doing anything random :laughing: